connections.py

```diff
@@ -63,14 +64,33 @@ def get_buckets(self, vehicle_id, data_type, station):
             for b in bucket_list:
                 bucket_list[b] = round(100 * (bucket_list[b] / number_of_entries), PRECISION)
 
+            print(bucket_list)
+
             return bucket_list
         else:
             return None
 
+    def handle_invalid_vehicle_id(self, vehicle_id):
+        # S trains are mapped on L trains in the data
+        # Train number are always the last 4 digits, but we have S51 and S1
+        # trains for example
+        if 'S' in vehicle_id:
+            vehicle_id = 'L' + vehicle_id[-4:]
```

Comment:
Would be cleaner to do `.replace('S', 'L')` in case there are IDs with different lengths (and for future proofing)

Reply:
S trains have a number, this 'feature' doesn't allow us to do it with `replace`